lib/src/kbasesearchengine/main/IndexerWorker.java

```diff
@@ -844,15 +815,112 @@ private void indexObjectWrapperFn(final List<Object> input)
         
         private List<ObjectData> getObjectsByIds(final Set<GUID> guids)
                 throws RetriableIndexingException {
-            kbasesearchengine.search.PostProcessing pp = 
-                    new kbasesearchengine.search.PostProcessing();
-            pp.objectData = false;
-            pp.objectKeys = true;
-            try {
-                return indexingStorage.getObjectsByIds(guids, pp);
-            } catch (IOException e) {
-                throw new RetriableIndexingException(ErrorType.OTHER, e.getMessage(), e);
+
+            final List<ObjectData> data = new ArrayList<>();
+
+            // get object from data source
+            for (GUID guid: guids) {
+                final File tempFile;
+                try {
+                    FileUtil.getOrCreateSubDir(rootTempDir, guid.getStorageCode());
+                    tempFile = File.createTempFile("ws_srv_response_", ".json");
+                } catch (IOException ex) {
+                    throw new RetriableIndexingException(ErrorType.OTHER, ex.getMessage(), ex);
+                }
+
+                try {
+                    final EventHandler handler = getEventHandler(guid);
+                    final Iterator<ResolvedReference> refs = handler.resolveReferences(null, guids).iterator();
```

Comment:
Why not just pull the RR from the cache? It would've been added in resolveRefs().

Reply:
I'm not seeing a cache of resolved references that I can use at this point in the code and I don't understand this whole concept of resolving references. I think a call to discuss this might help.

Reply:
The cache is here: https://github.com/kbase/KBaseSearchEngine/blob/26f74aee7e733d0908e3cd4a2eb3336f253beb87/lib/src/kbasesearchengine/main/IndexerWorker.java#L713

Reply:
There's two reasons for resolving references. Let's say data A has a reference to data B in some data source. That reference may or may not be valid and may or may not be specified with a permanent address, and so the ref resolving method contacts to the data source to a) ensure the ref is valid and b) translate the ref to a permanent address.

The second reason is that, at least in the case of the workspace, if an object is deleted the *only* way to access it is via a ref path, even for admins. There are technical reasons for this that I won't go into here. So if Genome G depends on Assembly A and assembly A is deleted, a the indexing recursion needs to supply a ref path from G to A so A can be accessed, parsed, and any relevant fields flattened into the ES document for G.

If this still isn't clear let's have a slack call this afternoon.

Reply:
I spent hours trying to use refResolvingCache from getObjectsByIds and found it doesn't work because the cache stores refpaths without subobjId info and getObjectByIds processes guids with subObjectIds. So when the cache lookup takes place, the resolved reference for a guid with subObjectId results in a reference to the main object and not the subObject. This eventually results in a guidNotFoundError because in KeywordParser,

```
final Map<GUID, SearchObjectType> guidToType = lookup.getTypesForGuids(objectRefPath, guids)
``` 
the guidToType contains the map for the main object and then fails on, 

```
if (!guidToType.containsKey(guid)) {
       throw new GUIDNotFoundException("GUID " + guid + " not found");
}
```
as the guid contains the subObjectId info.

Reply:
just resolve the main part of the GUID, not the subobject stuff.

Reply:
Done.